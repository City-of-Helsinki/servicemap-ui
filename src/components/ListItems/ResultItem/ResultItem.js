/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import { intlShape } from 'react-intl';
import {
  ListItem, ListItemIcon, Typography, Divider,
} from '@material-ui/core';

// TODO: Complete distance calculations and related accessibility texts
const ResultItem = ({
  bottomHighlight,
  bottomText,
  classes,
  onClick,
  onKeyDown,
  icon,
  subtitle,
  title,
  distance,
  divider,
  role,
  srLabel,
  intl,
  selected,
  padded,
  extendedClasses,
}) => {
  // Screen reader text
  const srText = `
    ${title || ''} 
    ${subtitle || ''} 
    ${distance
    ? `${distance.distance} ${distance.type === 'm'
      ? intl.formatMessage({ id: 'general.distance.meters' })
      : intl.formatMessage({ id: 'general.distance.kilometers' })}`
    : ''} 
    ${bottomText || ''} 
    ${srLabel || ''}
  `;

  const typographyClasses = (extendedClasses && extendedClasses.typography) || {};
  const listItemClasses = padded ? classes.paddedItem : classes.listItem;
  const listItemIconClasses = padded ? classes.listItemIconPadded : classes.listItemIcon;

  return (
    <>
      <ListItem
        selected={selected}
        button
        role={role || 'link'}
        component="li"
        tabIndex={0}
        onClick={onClick}
        onKeyDown={onKeyDown}
        className={listItemClasses}
      >
        {
          icon
          && (
          <ListItemIcon className={listItemIconClasses}>
            {icon}
          </ListItemIcon>
          )
        }
        <div className={classes.itemTextContainer}>
          <div className={classes.topRow || ''}>
            {
              // SROnly element with full readable text
            }
            <Typography
              className={`${classes.title || ''} ResultItem-srOnly`}
              component="p"
              variant="srOnly"
            >
              {srText}
            </Typography>

            {
              // Title
            }
            <Typography
              className={`${classes.title || ''}  ${typographyClasses.title || ''} ResultItem-title`}
              component="p"
              role="textbox"
              variant="body2"
              aria-hidden="true"
            >
              {title}
            </Typography>

            {
              // Distance text
              distance
              && (
                <div className={`${classes.rightColumn || ''}`}>
                  <Typography
                    variant="caption"
                    className={`${classes.caption || ''} ${classes.text} ${classes.marginLeft || ''} ${typographyClasses.topRight || ''} ResultItem-distance`}
                    component="p"
                    aria-hidden="true"
                  >
                    {distance.distance}
                    {distance.type}
                  </Typography>
                </div>
              )
            }

          </div>
          {
            // Bottom row
            (subtitle || bottomText)
            && (
              <div>
                <div>
                  <Typography
                    variant="caption"
                    className={`${classes.noMargin || ''} ${classes.text} ${typographyClasses.subtitle || ''} ResultItem-subtitle`}
                    component="p"
                    aria-hidden="true"
                  >
                    {subtitle || ''}
                  </Typography>
                </div>
                {
                  bottomText
                  && (
                  <div className={`${classes.bottomContainer} ${bottomHighlight ? classes.bottomHighlight : ''}`}>
                    <Typography
                      className={`${classes.smallFont || ''} ${typographyClasses.bottom || ''} ResultItem-bottom`}
                      color="inherit"
                      component="p"
                      variant="caption"
                      aria-hidden="true"
                    >
                      {bottomText}
                    </Typography>
                  </div>
                  )
                }

              </div>
            )
          }
        </div>
      </ListItem>
      {divider && (
        <li aria-hidden>
          <Divider className={classes.divider} variant="inset" />
        </li>
      )}
    </>
  );
};

export default ResultItem;

// Typechecking
ResultItem.propTypes = {
  bottomHighlight: PropTypes.bool,
  bottomText: PropTypes.string,
  classes: PropTypes.objectOf(PropTypes.any),
  extendedClasses: PropTypes.objectOf(PropTypes.shape({
    typography: PropTypes.shape({
      bottom: PropTypes.string,
      subtitle: PropTypes.string,
      title: PropTypes.string,
      topRight: PropTypes.string,
    }),
  })),
  icon: PropTypes.node,
  onClick: PropTypes.func,
  onKeyDown: PropTypes.func,
  subtitle: PropTypes.string,
  title: PropTypes.string.isRequired,
  distance: PropTypes.shape({
    distance: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    type: PropTypes.oneOf(['m', 'km']),
    text: PropTypes.string,
  }),
  divider: PropTypes.bool,
  role: PropTypes.string,
  srLabel: PropTypes.string,
  selected: PropTypes.bool,
  padded: PropTypes.bool,
  intl: intlShape.isRequired,
};

ResultItem.defaultProps = {
  bottomHighlight: false,
  bottomText: null,
  classes: {},
  extendedClasses: null,
  icon: null,
  onClick: () => {},
  onKeyDown: null,
  subtitle: null,
  distance: null,
  divider: true,
  role: null,
  srLabel: null,
  selected: false,
  padded: false,
};
