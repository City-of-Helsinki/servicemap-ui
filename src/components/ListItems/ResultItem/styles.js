
export default theme => ({
  listItemPadded: {
    padding: theme.spacing.unitDouble,
  },
  listItem: {
    padding: theme.spacing.unit,
  },
  listItemIcon: {
    margin: theme.spacing.unit,
    marginRight: theme.spacing.unitDouble,
  },
  listItemIconPadded: {
    margin: theme.spacing.unit,
  },
  bottomContainer: {
    display: 'inline-block',
    marginTop: '6px',
    '& p': {
      fontWeight: 'normal',
    },
  },
  bottomHighlight: {
    backgroundColor: '#323232',
    color: '#FFF',
    padding: '0 8px',
    marginBottom: '2px',
  },
  cssFocused: {
    outlineStyle: 'solid',
    outlineColor: 'blue',
    outlineWidth: 2,
  },
  title: {
    flex: '1 1 auto',
    textOverflow: 'ellipsis',
    margin: 0,
  },
  noMargin: {
    margin: 0,
  },
  itemTextContainer: {
    flex: '1 1 auto',
    margin: 0,
    marginLeft: theme.spacing.unitDouble,
  },
  topRow: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
  },
  bottomRow: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
  },
  marginLeft: {
    marginLeft: theme.spacing.unit,
  },
  rightColumn: {
    textAlign: 'right',
  },
  caption: {
    color: 'rgba(0,0,0,0.6)',
  },
  text: {
    color: '#000',
    fontWeight: 'normal',
  },
  divider: {
    marginRight: -theme.spacing.unitDouble,
  },
});
