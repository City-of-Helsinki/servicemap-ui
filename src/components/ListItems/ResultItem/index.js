import ResultItem from './ResultItem';

export default ResultItem;
