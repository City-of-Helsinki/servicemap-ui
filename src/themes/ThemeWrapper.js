import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ThemeProvider } from '@mui/material/styles';
import themes from '.';

const mapStateToProps = (state) => {
  const { user } = state;
  return {
    theme: user.theme,
  };
};

// Component to handle theme changes
const ThemeHandler = ({ children, theme }) => (
// Get correct theme setting from store
  <ThemeProvider theme={theme === 'dark' ? themes.SMThemeDark : themes.SMTheme}>
    {children}
  </ThemeProvider>
);
ThemeHandler.propTypes = {
  children: PropTypes.node,
  theme: PropTypes.string.isRequired,
};
ThemeHandler.defaultProps = {
  children: null,
};

const ThemeWrapper = connect(
  mapStateToProps,
)(ThemeHandler);

export default ThemeWrapper;
