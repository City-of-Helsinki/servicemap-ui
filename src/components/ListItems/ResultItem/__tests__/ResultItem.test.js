// Link.react.test.js
import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import { MuiThemeProvider } from '@material-ui/core';
import themes from '../../../../themes';
import ResultItem from '../index';

// Generic required props for ResultItem
const mockProps = {
  title: 'Title text',
  bottomText: 'Bottom text',
  distance: {
    text: '100m',
    srText: '100 metrin päässä',
  },
  subtitle: 'Subtitle text',
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <MuiThemeProvider theme={themes.SMTheme}>
    {children}
  </MuiThemeProvider>
);

describe('<ResultItem />', () => {
  // let render;
  let mount;

  beforeEach(() => {
    // render = createRender({ wrappingComponent: Providers });
    mount = createMount({ wrappingComponent: Providers });
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('should work', () => {
    const component = mount(<ResultItem {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('simulates click event', () => {
    const mockCallBack = jest.fn();
    const component = mount(<ResultItem {...mockProps} onClick={mockCallBack} />);

    component.find('ForwardRef(ListItem)').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('simulates keyboard event', () => {
    const mockCallBack = jest.fn();
    const component = mount(<ResultItem {...mockProps} onClick={mockCallBack} />);

    component.find('ForwardRef(ListItem)').simulate('keyDown', { which: 13 });
    component.find('ForwardRef(ListItem)').simulate('keyDown', { which: 32 });
    expect(mockCallBack.mock.calls.length).toEqual(2);
  });

  it('does show text correctly', () => {
    const component = mount(<ResultItem {...mockProps} />);

    const p = component.find('p');
    expect(p.at(1).text()).toEqual(mockProps.title);
    expect(p.at(2).text()).toEqual(mockProps.distance.text);
    expect(p.at(3).text()).toEqual(mockProps.subtitle);
    expect(p.at(4).text()).toEqual(mockProps.bottomText);
  });

  it('does set select correctly', () => {
    const component = mount(<ResultItem {...mockProps} selected />);
    expect(component.find('ForwardRef(ListItem)').props().selected).toBeTruthy();
  });

  it('does set divider correctly', () => {
    const component = mount(<ResultItem {...mockProps} />);
    expect(component.find('ForwardRef(Divider)').exists()).toBeTruthy();
  });

  it('does use default accessibility attributes correctly', () => {
    const component = mount(<ResultItem {...mockProps} />);

    // Expect screen reader texts to render correctly
    const srText = component.find('p.ResultItem-srOnly').text();
    const containsText = srText.indexOf(mockProps.title) !== -1
      && srText.indexOf(mockProps.subtitle) !== -1
      && srText.indexOf(mockProps.distance.srText) !== -1
      && srText.indexOf(mockProps.bottomText) !== -1;
    expect(containsText).toBeTruthy();

    // Expect aria-hidden attributes to be placed correctly
    const paragraphs = component.find('p');
    expect(paragraphs.at(0).props()['aria-hidden']).toBeFalsy();
    expect(paragraphs.at(1).props()['aria-hidden']).toEqual('true');
    expect(paragraphs.at(2).props()['aria-hidden']).toEqual('true');
    expect(paragraphs.at(3).props()['aria-hidden']).toEqual('true');
    expect(paragraphs.at(4).props()['aria-hidden']).toEqual('true');

    // Expect role to be set
    expect(component.find('ForwardRef(ListItem)').props().role).toEqual('link');
    
    // Expect element to have tabIndex 0
    expect(component.find('ForwardRef(ListItem)').props().tabIndex).toEqual(0);
  });

  it('does use given accessibility attributes correctly', () => {
    const component = mount(
      <ResultItem
        {...mockProps}
        role="button"
        srLabel="screen reader label"
      />,
    );

    const srText = component.find('p.ResultItem-srOnly').text();
    const containsText = srText.indexOf(mockProps.title) !== -1
      && srText.indexOf(mockProps.subtitle) !== -1
      && srText.indexOf(mockProps.distance.srText) !== -1
      && srText.indexOf(mockProps.bottomText) !== -1
      && srText.indexOf('screen reader label') !== -1;

    // Expect screen reader texts to render correctly
    expect(containsText).toBeTruthy();
    // Expect role to be set
    expect(component.props().role).toEqual('button');
  });
});
