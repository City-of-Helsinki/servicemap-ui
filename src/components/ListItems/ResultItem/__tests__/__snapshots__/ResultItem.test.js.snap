// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultItem /> should work 1`] = `
<WithStyles(ResultItem)
  bottomText="Bottom text"
  distance={
    Object {
      "srText": "100 metrin päässä",
      "text": "100m",
    }
  }
  subtitle="Subtitle text"
  title="Title text"
>
  <ResultItem
    bottomHighlight={false}
    bottomText="Bottom text"
    classes={
      Object {
        "bottomContainer": "ResultItem-bottomContainer-5",
        "bottomHighlight": "ResultItem-bottomHighlight-6",
        "bottomRow": "ResultItem-bottomRow-11",
        "caption": "ResultItem-caption-14",
        "divider": "ResultItem-divider-16",
        "itemTextContainer": "ResultItem-itemTextContainer-9",
        "listItem": "ResultItem-listItem-2",
        "listItemIcon": "ResultItem-listItemIcon-3",
        "listItemIconPadded": "ResultItem-listItemIconPadded-4",
        "listItemPadded": "ResultItem-listItemPadded-1",
        "marginLeft": "ResultItem-marginLeft-12",
        "noMargin": "ResultItem-noMargin-8",
        "rightColumn": "ResultItem-rightColumn-13",
        "text": "ResultItem-text-15",
        "title": "ResultItem-title-7",
        "topRow": "ResultItem-topRow-10",
      }
    }
    distance={
      Object {
        "srText": "100 metrin päässä",
        "text": "100m",
      }
    }
    divider={true}
    extendedClasses={null}
    icon={null}
    onClick={[Function]}
    padded={false}
    role="link"
    selected={false}
    srLabel={null}
    subtitle="Subtitle text"
    title="Title text"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      className="ResultItem-listItem-2"
      component="li"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      selected={false}
      tabIndex={0}
    >
      <ForwardRef(ListItem)
        button={true}
        className="ResultItem-listItem-2"
        classes={
          Object {
            "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
            "button": "MuiListItem-button",
            "container": "MuiListItem-container",
            "dense": "MuiListItem-dense",
            "disabled": "Mui-disabled",
            "divider": "MuiListItem-divider",
            "focusVisible": "Mui-focusVisible",
            "gutters": "MuiListItem-gutters",
            "root": "MuiListItem-root",
            "secondaryAction": "MuiListItem-secondaryAction",
            "selected": "Mui-selected",
          }
        }
        component="li"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="link"
        selected={false}
        tabIndex={0}
      >
        <WithStyles(ForwardRef(ButtonBase))
          className="MuiListItem-root ResultItem-listItem-2 MuiListItem-gutters MuiListItem-button"
          component="li"
          disabled={false}
          focusVisibleClassName="Mui-focusVisible"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="link"
          tabIndex={0}
        >
          <ForwardRef(ButtonBase)
            className="MuiListItem-root ResultItem-listItem-2 MuiListItem-gutters MuiListItem-button"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            component="li"
            disableRipple={true}
            disabled={false}
            focusVisibleClassName="Mui-focusVisible"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="link"
            tabIndex={0}
          >
            <li
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root ResultItem-listItem-2 MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="link"
              tabIndex={0}
            >
              <div
                className="ResultItem-itemTextContainer-9"
                key=".1"
              >
                <div
                  className="ResultItem-topRow-10"
                >
                  <WithStyles(ForwardRef(Typography))
                    className="ResultItem-title-7 ResultItem-srOnly"
                    component="p"
                    variant="srOnly"
                  >
                    <ForwardRef(Typography)
                      className="ResultItem-title-7 ResultItem-srOnly"
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter",
                          "alignJustify": "MuiTypography-alignJustify",
                          "alignLeft": "MuiTypography-alignLeft",
                          "alignRight": "MuiTypography-alignRight",
                          "body1": "MuiTypography-body1",
                          "body2": "MuiTypography-body2",
                          "button": "MuiTypography-button",
                          "caption": "MuiTypography-caption",
                          "colorError": "MuiTypography-colorError",
                          "colorInherit": "MuiTypography-colorInherit",
                          "colorPrimary": "MuiTypography-colorPrimary",
                          "colorSecondary": "MuiTypography-colorSecondary",
                          "colorTextPrimary": "MuiTypography-colorTextPrimary",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary",
                          "displayBlock": "MuiTypography-displayBlock",
                          "displayInline": "MuiTypography-displayInline",
                          "gutterBottom": "MuiTypography-gutterBottom",
                          "h1": "MuiTypography-h1",
                          "h2": "MuiTypography-h2",
                          "h3": "MuiTypography-h3",
                          "h4": "MuiTypography-h4",
                          "h5": "MuiTypography-h5",
                          "h6": "MuiTypography-h6",
                          "noWrap": "MuiTypography-noWrap",
                          "overline": "MuiTypography-overline",
                          "paragraph": "MuiTypography-paragraph",
                          "root": "MuiTypography-root",
                          "srOnly": "MuiTypography-srOnly",
                          "subtitle1": "MuiTypography-subtitle1",
                          "subtitle2": "MuiTypography-subtitle2",
                        }
                      }
                      component="p"
                      variant="srOnly"
                    >
                      <p
                        className="MuiTypography-root ResultItem-title-7 ResultItem-srOnly MuiTypography-srOnly"
                      >
                        
    Title text 
    Subtitle text 
    100 metrin päässä 
    Bottom text 
    
  
                      </p>
                    </ForwardRef(Typography)>
                  </WithStyles(ForwardRef(Typography))>
                  <WithStyles(ForwardRef(Typography))
                    aria-hidden="true"
                    className="ResultItem-title-7   ResultItem-title"
                    component="p"
                    role="textbox"
                    variant="body2"
                  >
                    <ForwardRef(Typography)
                      aria-hidden="true"
                      className="ResultItem-title-7   ResultItem-title"
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter",
                          "alignJustify": "MuiTypography-alignJustify",
                          "alignLeft": "MuiTypography-alignLeft",
                          "alignRight": "MuiTypography-alignRight",
                          "body1": "MuiTypography-body1",
                          "body2": "MuiTypography-body2",
                          "button": "MuiTypography-button",
                          "caption": "MuiTypography-caption",
                          "colorError": "MuiTypography-colorError",
                          "colorInherit": "MuiTypography-colorInherit",
                          "colorPrimary": "MuiTypography-colorPrimary",
                          "colorSecondary": "MuiTypography-colorSecondary",
                          "colorTextPrimary": "MuiTypography-colorTextPrimary",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary",
                          "displayBlock": "MuiTypography-displayBlock",
                          "displayInline": "MuiTypography-displayInline",
                          "gutterBottom": "MuiTypography-gutterBottom",
                          "h1": "MuiTypography-h1",
                          "h2": "MuiTypography-h2",
                          "h3": "MuiTypography-h3",
                          "h4": "MuiTypography-h4",
                          "h5": "MuiTypography-h5",
                          "h6": "MuiTypography-h6",
                          "noWrap": "MuiTypography-noWrap",
                          "overline": "MuiTypography-overline",
                          "paragraph": "MuiTypography-paragraph",
                          "root": "MuiTypography-root",
                          "srOnly": "MuiTypography-srOnly",
                          "subtitle1": "MuiTypography-subtitle1",
                          "subtitle2": "MuiTypography-subtitle2",
                        }
                      }
                      component="p"
                      role="textbox"
                      variant="body2"
                    >
                      <p
                        aria-hidden="true"
                        className="MuiTypography-root ResultItem-title-7   ResultItem-title MuiTypography-body2"
                        role="textbox"
                      >
                        Title text
                      </p>
                    </ForwardRef(Typography)>
                  </WithStyles(ForwardRef(Typography))>
                  <div
                    className="ResultItem-rightColumn-13"
                  >
                    <WithStyles(ForwardRef(Typography))
                      aria-hidden="true"
                      className="ResultItem-caption-14 ResultItem-text-15 ResultItem-marginLeft-12  ResultItem-distance"
                      component="p"
                      variant="caption"
                    >
                      <ForwardRef(Typography)
                        aria-hidden="true"
                        className="ResultItem-caption-14 ResultItem-text-15 ResultItem-marginLeft-12  ResultItem-distance"
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter",
                            "alignJustify": "MuiTypography-alignJustify",
                            "alignLeft": "MuiTypography-alignLeft",
                            "alignRight": "MuiTypography-alignRight",
                            "body1": "MuiTypography-body1",
                            "body2": "MuiTypography-body2",
                            "button": "MuiTypography-button",
                            "caption": "MuiTypography-caption",
                            "colorError": "MuiTypography-colorError",
                            "colorInherit": "MuiTypography-colorInherit",
                            "colorPrimary": "MuiTypography-colorPrimary",
                            "colorSecondary": "MuiTypography-colorSecondary",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary",
                            "displayBlock": "MuiTypography-displayBlock",
                            "displayInline": "MuiTypography-displayInline",
                            "gutterBottom": "MuiTypography-gutterBottom",
                            "h1": "MuiTypography-h1",
                            "h2": "MuiTypography-h2",
                            "h3": "MuiTypography-h3",
                            "h4": "MuiTypography-h4",
                            "h5": "MuiTypography-h5",
                            "h6": "MuiTypography-h6",
                            "noWrap": "MuiTypography-noWrap",
                            "overline": "MuiTypography-overline",
                            "paragraph": "MuiTypography-paragraph",
                            "root": "MuiTypography-root",
                            "srOnly": "MuiTypography-srOnly",
                            "subtitle1": "MuiTypography-subtitle1",
                            "subtitle2": "MuiTypography-subtitle2",
                          }
                        }
                        component="p"
                        variant="caption"
                      >
                        <p
                          aria-hidden="true"
                          className="MuiTypography-root ResultItem-caption-14 ResultItem-text-15 ResultItem-marginLeft-12  ResultItem-distance MuiTypography-caption"
                        >
                          100m
                        </p>
                      </ForwardRef(Typography)>
                    </WithStyles(ForwardRef(Typography))>
                  </div>
                </div>
                <div>
                  <div>
                    <WithStyles(ForwardRef(Typography))
                      aria-hidden="true"
                      className="ResultItem-noMargin-8 ResultItem-text-15  ResultItem-subtitle"
                      component="p"
                      variant="caption"
                    >
                      <ForwardRef(Typography)
                        aria-hidden="true"
                        className="ResultItem-noMargin-8 ResultItem-text-15  ResultItem-subtitle"
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter",
                            "alignJustify": "MuiTypography-alignJustify",
                            "alignLeft": "MuiTypography-alignLeft",
                            "alignRight": "MuiTypography-alignRight",
                            "body1": "MuiTypography-body1",
                            "body2": "MuiTypography-body2",
                            "button": "MuiTypography-button",
                            "caption": "MuiTypography-caption",
                            "colorError": "MuiTypography-colorError",
                            "colorInherit": "MuiTypography-colorInherit",
                            "colorPrimary": "MuiTypography-colorPrimary",
                            "colorSecondary": "MuiTypography-colorSecondary",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary",
                            "displayBlock": "MuiTypography-displayBlock",
                            "displayInline": "MuiTypography-displayInline",
                            "gutterBottom": "MuiTypography-gutterBottom",
                            "h1": "MuiTypography-h1",
                            "h2": "MuiTypography-h2",
                            "h3": "MuiTypography-h3",
                            "h4": "MuiTypography-h4",
                            "h5": "MuiTypography-h5",
                            "h6": "MuiTypography-h6",
                            "noWrap": "MuiTypography-noWrap",
                            "overline": "MuiTypography-overline",
                            "paragraph": "MuiTypography-paragraph",
                            "root": "MuiTypography-root",
                            "srOnly": "MuiTypography-srOnly",
                            "subtitle1": "MuiTypography-subtitle1",
                            "subtitle2": "MuiTypography-subtitle2",
                          }
                        }
                        component="p"
                        variant="caption"
                      >
                        <p
                          aria-hidden="true"
                          className="MuiTypography-root ResultItem-noMargin-8 ResultItem-text-15  ResultItem-subtitle MuiTypography-caption"
                        >
                          Subtitle text
                        </p>
                      </ForwardRef(Typography)>
                    </WithStyles(ForwardRef(Typography))>
                  </div>
                  <div
                    className="ResultItem-bottomContainer-5 "
                  >
                    <WithStyles(ForwardRef(Typography))
                      aria-hidden="true"
                      className="  ResultItem-bottom"
                      color="inherit"
                      component="p"
                      variant="caption"
                    >
                      <ForwardRef(Typography)
                        aria-hidden="true"
                        className="  ResultItem-bottom"
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter",
                            "alignJustify": "MuiTypography-alignJustify",
                            "alignLeft": "MuiTypography-alignLeft",
                            "alignRight": "MuiTypography-alignRight",
                            "body1": "MuiTypography-body1",
                            "body2": "MuiTypography-body2",
                            "button": "MuiTypography-button",
                            "caption": "MuiTypography-caption",
                            "colorError": "MuiTypography-colorError",
                            "colorInherit": "MuiTypography-colorInherit",
                            "colorPrimary": "MuiTypography-colorPrimary",
                            "colorSecondary": "MuiTypography-colorSecondary",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary",
                            "displayBlock": "MuiTypography-displayBlock",
                            "displayInline": "MuiTypography-displayInline",
                            "gutterBottom": "MuiTypography-gutterBottom",
                            "h1": "MuiTypography-h1",
                            "h2": "MuiTypography-h2",
                            "h3": "MuiTypography-h3",
                            "h4": "MuiTypography-h4",
                            "h5": "MuiTypography-h5",
                            "h6": "MuiTypography-h6",
                            "noWrap": "MuiTypography-noWrap",
                            "overline": "MuiTypography-overline",
                            "paragraph": "MuiTypography-paragraph",
                            "root": "MuiTypography-root",
                            "srOnly": "MuiTypography-srOnly",
                            "subtitle1": "MuiTypography-subtitle1",
                            "subtitle2": "MuiTypography-subtitle2",
                          }
                        }
                        color="inherit"
                        component="p"
                        variant="caption"
                      >
                        <p
                          aria-hidden="true"
                          className="MuiTypography-root   ResultItem-bottom MuiTypography-caption MuiTypography-colorInherit"
                        >
                          Bottom text
                        </p>
                      </ForwardRef(Typography)>
                    </WithStyles(ForwardRef(Typography))>
                  </div>
                </div>
              </div>
            </li>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(ListItem)>
    </WithStyles(ForwardRef(ListItem))>
    <li
      aria-hidden={true}
    >
      <WithStyles(ForwardRef(Divider))
        className="ResultItem-divider-16"
        variant="inset"
      >
        <ForwardRef(Divider)
          className="ResultItem-divider-16"
          classes={
            Object {
              "absolute": "MuiDivider-absolute",
              "flexItem": "MuiDivider-flexItem",
              "inset": "MuiDivider-inset",
              "light": "MuiDivider-light",
              "middle": "MuiDivider-middle",
              "root": "MuiDivider-root",
              "vertical": "MuiDivider-vertical",
            }
          }
          variant="inset"
        >
          <hr
            className="MuiDivider-root ResultItem-divider-16 MuiDivider-inset"
          />
        </ForwardRef(Divider)>
      </WithStyles(ForwardRef(Divider))>
    </li>
  </ResultItem>
</WithStyles(ResultItem)>
`;
